fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -687,9 +760,28 @@ private CompletableFuture<Void> moveDocsFromPartition(@Nonnull final LuceneParti
      * @param groupingKey grouping key
      * @return future list of partition metadata
      */
-    private CompletableFuture<List<LucenePartitionInfoProto.LucenePartitionInfo>> getAllPartitionMetaInfo(@Nonnull final Tuple groupingKey) {
-        Range range = state.indexSubspace.subspace(Tuple.from(groupingKey, PARTITION_META_SUBSPACE)).range();
+    @VisibleForTesting
+    public CompletableFuture<List<LucenePartitionInfoProto.LucenePartitionInfo>> getAllPartitionMetaInfo(@Nonnull final Tuple groupingKey) {
+        Range range = state.indexSubspace.subspace(groupingKey.add(PARTITION_META_SUBSPACE)).range();
         final AsyncIterable<KeyValue> rangeIterable = state.context.ensureActive().getRange(range, Integer.MAX_VALUE, true, StreamingMode.WANT_ALL);
         return AsyncUtil.collect(rangeIterable).thenApply(all -> all.stream().map(LucenePartitioner::partitionInfoFromKV).collect(Collectors.toList()));
     }
+
+    public static CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> getNextPartitionInfo(
```

Comment:
This is needed for `FDBDirectoryManager` to find the partitions to do merging.
I think this adds to the argument that the relationship between the layers here is messy and needs to be rethought.